Cleanup / Update Coder Eyetracking demos

isolver opened the pull request:
RF: All eye tracking coder demo's now use the window arg with launchHubServer()
RF: All eye tracking coder demo's now use MouseGaze ( 'mouse' ) as default eye tracker.
RF: gcCursor demo now uses dicts for eye tracker config to match other demos. 
RF: No eye tracker demo's use / need an iohub_config.yaml.
FF/DEL: Removed redundant eye tracking validation demo.
DEL: Removed selectTracker demo. Now that there is Builder integration, it should be used for this type of application.
RF: Updated iohub display device default config docstr and default for using PsychoPy Monitor config file.

isolver committed:
Removed unneeded prints

Removed unneeded prints

isolver committed:
FF: Removed redundant eye tracker validation demo

FF: Removed redundant eye tracker validation demo
RF: Cleanup of simple.py eye tracker demo. Default tracker set to 'mouse'.

isolver committed:
RF: Cleanup and trimming of eye tracking coder demos

RF: Make MouseGaze default eye tracker in coder demos
RF: Do not save hdf5 file in simple.py demo
RF: Remove redundant call to win.winHandle.set_fullscreen(False)
DOC: Updated iohub display device default_config.yaml docstr and defaults
DEL: Removed selectTracker demo. Intent is now better served as a Builder demo and gcCursor demo fills the Coder need.
RF: Switch gcCursor demo to use dict for eye tracker config instead of default_config.yaml. Now no iohub eye tracking demo's use a iohub_config.yaml.
RF: All Coder iohub eye tracking demo's now use new 2021.2 launchHubServer args.

peircej commented:
For this PR the tests are currently failing with:
```
=================================== FAILURES ===================================
___________________________ testDefaultServerLaunch ____________________________

    def testDefaultServerLaunch():
        """
        """
        io = startHubProcess()
    
        # check that a kb and mouse have been created
>       keyboard=io.devices.keyboard
E       AttributeError: 'ioHubDevices' object has no attribute 'keyboard'

psychopy/tests/test_iohub/test_launch.py:12: AttributeError
----------------------------- Captured stdout call -----------------------------
ERROR: DEVICE CONFIG ERRORS FOUND! IOHUB NOT LOADING DEVICE: psychopy.iohub.devices.display
errors count 2:
	('reporting_unit_type', None)
	('color_space', None)


## Device was not started by the ioHub Server: Display
Error during device creation ....
Traceback (most recent call last):
  File "/Users/runner/hostedtoolcache/Python/3.6.13/x64/lib/python3.6/site-packages/psychopy/iohub/server.py", line 674, in createNewMonitoredDevice
Error:     raise ioHubError('Device config validation failed')
psychopy.iohub.errors.ioHubError: ioHubError:
Args: ('Device config validation failed',)
```

isolver committed:
FF: Allow iohub Monitor unit type to be None

FF: Allow iohub Monitor unit type to be None for headless (windowless) use of launchHubServer
